### ISSUE TITLE
Zero-RPM Speeds (Switch to 'Automatic' Control below lower Threshold)

### ISSUE THREAD

**Issue by WhatIsThisAllAboutSilly** [crystalidea/macs-fan-control#582]
My M1 Max MBP 16" when in automatic mode (I.e. MFC not controlling the fans) sets the fan speed to zero RPM.

MFC sets a minimum of 1500 RPM.

Can a feature be implemented that allows the zero RPM mode to be enabled?

Either setting the control mode to "Auto" when below the lower temp threshold, or setting a zero RPM speed below the threshold?

Thanks!

Edit: This would be a feature that you could add to "Pro" and I would purchase after this addition.

**Comment by kleuter** [comment]
I also own a mac studio and in Auto mode the fan is always at around 1320 rpm, current workaround is setting it at 1100 (noticebly quiter)
<img width="1116" alt="Screenshot 2022-03-30 at 09 27 41" src="https://user-images.githubusercontent.com/2600624/160775923-225e5c14-a290-487f-a0c4-6c9522861a29.png">

**Comment by J-charles-C** [comment]
I,

Same problem of thomam04.

Can you add this feature ?

best

**Comment by shp7724** [comment]
Can I get any update on this feature request?
I'm hearing the high pitch whine all the time that Mac Studio generates.
Setting rpm to ~1100 helps a bit, but does not solve the issue.
Would be really glad if the patch is out any time soon.

**Comment by kleuter** [comment]
In progress guys, first beta next week, thanks

**Comment by dirkschreib** [comment]
> In progress guys, first beta next week, thanks

Any news?

**Comment by kleuter** [comment]
I'm really sorry for this delay. Please be patient, I'm going to implement later this month.

**Comment by donnie-darko** [comment]
> I would support this feature request. I would like to be able to overwrite / change the minimum RPM setting to something lower than the system default.
> 
> E.g my Mac Studio has a minimum fan speed of 1100rpm but it's still not silent at idle / low CPU intensive workload.

I would love this new feature as far as it enable to overwrite the min RPM setting to lower than the sys default. 
I'm on iMac 24 M1 and here the min fan speed at idle is ~2500rpm but it's is not silent at all. I guess ~2000rpm will be much better.

Thanks!

**Comment by kleuter** [comment]
thanks, donnie for the idea. I also think that 2500 is too much here, will improve that somehow.

**Comment by zcjerry229** [comment]
Any ETA for this one?
Right now setting 1100 rpm for my Mac Studio is much quieter than 1300 rpm.
Looking forward to set rpm around 800 rpm after this change.

**Comment by dirkschreib** [comment]
Is there any news on this feature? Do you still think it is possible to go below 1100 rpm?
I‘m on my second Studio and this was initially quieter than the first one. But now the coil whine is back. 
Have to decide if I keep it or switch to a MacBook Pro.

**Comment by shp7724** [comment]
Tired of waiting, I ended up tweaking [smcFanControl](https://github.com/hholtmann/smcFanControl) and putting it into my own Alfred Workflow. (If you want to know how, just refer to the PR of the repo.)

According to my experiment, in Mac Studio, it is possible to set the fan speed to 0 rpm, or 500 rpm and above. However, after a few hours of turning the fan off, I noticed an increase of temperature up to 60°C even when idle. When set to 500 rpm or above, the coil whine did get quieter, but not silent, unfortunately. In the end, I stopped manually adjusting the fan speed and decided to get used to the coil whine.

I'm leaving this comment just in case you are curious.

**Comment by kleuter** [comment]
This feature is already under development, will be shipped soon as a beta.

**shp7724** thanks for the follow up. 60°C is not a great deal actually. The main question is: what value should the app **safely** allow to set manually ? 500, 700, 900... 1000 rpm?

**Comment by shp7724** [comment]
@kleuter You are right, but the fact that the heat is trapped in the body makes me worry about any potential damage it might cause in the long term.

I'm not sure what the safe limit should be. Personally I wished I could set it to 0 rpm when idle, 500rpm on intermediate workload, 1300rpm on high workload, and so on. In fact, that's what MacBook with the same chip does already.

**Comment by junxiao87** [comment]
60 ℃ would be safe for the chips.first we need the 0rpm to eliminate the whinethen we can tune the autofan control curve later. 
    	Get Outlook for iOS
    
  

 From: ***@***.*** ***@***.***> on behalf of shinhong_park ***@***.***>Sent: Sunday, June 26, 2022 10:00 AMTo: crystalidea/macs-fan-control ***@***.***>Cc: junxiao87 ***@***.***>; Manual ***@***.***>Subject: Re: [crystalidea/macs-fan-control] Zero-RPM Speeds (Switch to 'Automatic' Control below lower Threshold) (Issue #582) 
@kleuter You are right, but the fact that the heat is trapped in the body makes me worry about any potential damage it might cause in the long term.
I'm not sure what the safe limit should be. Personally I wished I could set it to 0 rpm when idle, 500rpm on intermediate workload, 1300rpm on high workload, and so on. In fact, that's what MacBook with the same chip does already.

**Comment by thomam04** [comment]
I don't think there is a need to set the fans in the Mac Studio below 900rpm. That should be completely inaudible and still save - (provides some sort of steady air flow). 

@shp7724 Try to cover part of the back exhaust port. I have taped off about 2/3 (starting from the top) - which greatly reduced the fan noise. I don't have coil whine, though.

**Comment by shp7724** [comment]
@thomam04 The fan noise itself isn't really a problem for me, but the whine is. It does not disappear unless the fan is completely turned off. I think that's why people are requesting for the zero rpm feature.

**Comment by modemm** [comment]
As I am running home recording studio, I would really appreciate to have 0 rpm option as this would be great to temporarily decrease it to zero during the take and then put it back on to let's say 500 rpm when editing and mixing

**Comment by KGB44** [comment]
> I don't think there is a need to set the fans in the Mac Studio below 900rpm. That should be completely inaudible and still save - (provides some sort of steady air flow).
> 
> @shp7724 Try to cover part of the back exhaust port. I have taped off about 2/3 (starting from the top) - which greatly reduced the fan noise. I don't have coil whine, though.

That's wrong. A stopped Fan is M1 Max and M1 Ultra must have since the MacBook Pro, Air and MacBook is able to do that  . The Fan is noisy at 1100 rpm as well as standard config 1300 rpm

**Comment by KGB44** [comment]
> This feature is already under development, will be shipped soon as a beta.
> 
> **shp7724** thanks for the follow up. 60°C is not a great deal actually. The main question is: what value should the app **safely** allow to set manually ? 500, 700, 900... 1000 rpm

> As I am running home recording studio, I would really appreciate to have 0 rpm option as this would be great to temporarily decrease it to zero during the take and then put it back on to let's say 500 rpm when editing and mixing
> I would say that even long term 60 degrees is still normal and ok. I also run mac mini 2020 with core i5 and I mostly have temps around 50-60 when idle.



> This feature is already under development, will be shipped soon as a beta.
> 
> **shp7724** thanks for the follow up. 60°C is not a great deal actually. The main question is: what value should the app **safely** allow to set manually ? 500, 700, 900... 1000 rpm?

manually allow set to "0" with a warning hint or a "professional setting"

**Comment by KGB44** [comment]
> Tired of waiting, I ended up tweaking [smcFanControl](https://github.com/hholtmann/smcFanControl) and putting it into my own Alfred Workflow. (If you want to know how, just refer to the PR of the repo.)
> 
> According to my experiment, in Mac Studio, it is possible to set the fan speed to 0 rpm, or 500 rpm and above. However, after a few hours of turning the fan off, I noticed an increase of temperature up to 60°C even when idle. When set to 500 rpm or above, the coil whine did get quieter, but not silent, unfortunately. In the end, I stopped manually adjusting the fan speed and decided to get used to the coil whine.
> 
> I'm leaving this comment just in case you are curious.

I've downloaded smcfancontrol 2.6, but I can't regulate the fan speed on m1 max Mac studio ...

**Comment by junxiao87** [comment]
Will Ver1.6 come in early of this Aug?

**Comment by dirkschreib** [comment]
> In progress guys, first beta next week, thanks

I don‘t want to put pressure on anyone, but what happened since this quote from April?
For me 1100 rpm is better than ~1300 rpm most of the time, sometimes the whine is „uglier“. 
I think, 0 rpm should be possible without overheating.

**Comment by dirkschreib** [comment]
Ok, I used the link from @shp7724 and the following four statements to silence my Mac Studio.
(All the usual disclaimers apply here. Do it on your own risk. May cause overheating or whatever.)
sudo ./smc -k "F0Md" -w 01
sudo ./smc -k "F1Md" -w 01
sudo ./smc -k "F0Tg" -w 00000000
sudo ./smc -k "F1Tg" -w 00000000

To turn the fans back on:
sudo ./smc -k "F0Md" -w 00
sudo ./smc -k "F1Md" -w 00

**Comment by KGB44** [comment]
> 

Thanks! :) I've installed brew and smcfancontrol. It's running, but in terminal the commands are not found...

**Comment by dirkschreib** [comment]
I compiled it (including PR 137 for Apple Silicon support) on another Mac and copied only one file to the Mac Studio. If you installed the GUI-App in the „normal“ folder, smc should be here: /Applications/smcFanControl.app/Contents/Resources/smc

**Comment by KGB44** [comment]
> I compiled it (including PR 137 for Apple Silicon support) on another Mac and copied only one file to the Mac Studio. If you installed the GUI-App in the „normal“ folder, smc should be here: /Applications/smcFanControl.app/Contents/Resources/smc

Amazing. It works. What a mess that the Mac Studio isn't stopping the fans in original. It neeeeeeeds to implemented into MacsFanControl soooo much. Wow, that's silent :D

**Comment by dirkschreib** [comment]
I made a Swift script without any external dependencies to automatically set the rpm values based on the temperature from one sensor (currently I use TT0D i.e. the first thunderbolt port). https://github.com/dirkschreib/SilentStudio

**Comment by modemm** [comment]
After couple of months of mac studio ownership I can say that when doing ordinary stuff, no fan needs to be turned on. So far never exceeded temp higher than 65°C. Stable idle temperature is around 45°C

**Comment by dirkschreib** [comment]
I‘m still testing various configurations. Currently I use „sudo ./SilentStudio.swift 0 900 57 AUTO“ and my Mac Studio has never reached 57°C. (BTW: This configuration will stay in AUTO mode forever if it is reached once. To get back to 900 rpm if temperature falls below 50°C use „sudo ./SilentStudio.swift 0 900 50 900 57 AUTO“.

**Comment by KGB44** [comment]
Cool and nice work! @dirkschreib ! How I can add this script into "Autostart" with starting my Mac? Any Ideas?

**Comment by dirkschreib** [comment]
Have not tried to autostart this script myself, but here are a few ideas: https://superuser.com/questions/229773/run-command-on-startup-login-mac-os-x. I would probably try a bash/zsh startup script first and then configure Terminal.app to start on logon.

**Comment by Anthonyzou** [comment]
Zero rpm mode is helpful because I already have a fan under my MacBook for extra cooling. And it manages under 50C most of the time without the external fan.

**Comment by MiroIsNot1337** [comment]
I bought Macs Fan Control hoping to tame the beast that is the Mac Studio's fan, but 1100rpm doesn't cut it for me as it is sitting only about 50cm from me on my desk. I downloaded your script. @dirkschreib but I am not clever enough to actually use it. I got as far as opening the terminal, dragging SilentStudio.swift into it, adding sudo in front of it and then... command not found. 

This is what it looks like: miro@Mac-Studio ~ % sudo ./Users/miro/Desktop/SilentStudio.swift -h

The mistake is probably as obvious as the solution, but it was already an adventure for me to download the .swift file from GitHub. Can someone help me or point me in the right direction?

My failed attempts for your entertainment:
<img width="581" alt="Screenshot 12 01 2023 um 12 04 27 PM" src="https://user-images.githubusercontent.com/122529727/212050879-7bff4e13-1bed-40cb-bd62-dc5d265505be.png">

**Comment by dirkschreib** [comment]
I don't know to which folder you have downloaded the script. Look it up in Finder and use that path. If it is on your desktop (as shown in your attempts) you have to make it executable with "chmod +x /Users/miro/Desktop/SilentStudio.swift".

**Comment by MiroIsNot1337** [comment]
Thank you, it works. This makes me so happy! Next, I'll see how I can add it to startup and how I can make sure it works after standby. I hope macs fan control will finally get the promised update to make this a bit more user-friendly, but I thank you for your script!

**Comment by MiroIsNot1337** [comment]
> I‘m still testing various configurations. Currently I use „sudo ./SilentStudio.swift 0 900 57 AUTO“ and my Mac Studio has never reached 57°C. (BTW: This configuration will stay in AUTO mode forever if it is reached once. To get back to 900 rpm if temperature falls below 50°C use „sudo ./SilentStudio.swift 0 900 50 900 57 AUTO“.

Apologies to bug you again. Do you have any idea why the fan turns on after standby and only quitting and restarting the script enforces my settings? 

sudo /Users/miro/Library/CloudStorage/Dropbox/portable\ software/SilentStudio.swift 0 0 57 500 60 900 65 AUTO 63 900 59 500 56 0

**Comment by dirkschreib** [comment]
„Apple“ thinks they have shut down the fans at sleep time. So it’s natural to wake them up after sleep. ;-)
I may incorporate sleep/wake detection in an app that uses SilentStudio.swift but would prefer to discuss this there and not to hijack this thread any further.

**Comment by MiroIsNot1337** [comment]
You're absolutely right. I have followed you and will comments in the respective issues in your repository.

**Comment by Passw0rdMayBeCompromised** [comment]
Just to add my two cents, I too would would love a feature that sets the fans at 0 (or 1100 RPM) but reverts to automatic if the temperature reaches X.
My Mac Studio also has the terrible whine so setting the fan to 1100 RPM solves that problem but I do not want to constantly turn the control off/on when I am launching a game or rendering something.

Basically, I want to avoid using the 1200-1500 RPM range.

Also, thanks for the App, I just bought it!

**Comment by MiroIsNot1337** [comment]
You can take a look at Dirk's script. He has even turned it into an app and you can tell it exactly how to control the fan. Mine is now turned off almost all of the time and will only turn on briefly while rendering a video or similar tasks.

**Comment by Passw0rdMayBeCompromised** [comment]
Thanks!
I might try it, if the function does get put into Macs Fan Control :)

**Comment by PAPPL** [comment]
Any updates to set Macs Fan Control to 0rpm if CPU-Temp is e.g. <60°C?

**Comment by MiroIsNot1337** [comment]
I suggest you take a look at the free SilentStudio. I purchased Macs Fan Control, but the feature was never implemented. Until it's finally added, SilentStudio is the only software that can do this for you while remaining fairly user friendly (to my knowledge).

**Comment by WhatIsThisAllAboutSilly** [comment]
Do you have a date for 1.6? Just realised I raised this 18 months ago 😯

**Comment by WhatIsThisAllAboutSilly** [comment]
For the record iStat Menu's 7 now does a nice fan curve that starts at 0%.

I've removed Macs Fan Control for now, but if iStat fan control works, then that as MFC done for me sadly.

**Comment by MiroIsNot1337** [comment]
I've tried it out and while I really like iStat Menu and might actually buy it for its other features, the fan curve doesn't work for me. At least not on my Mac Studio. Changing the fan curve doesn't do anything. The fan just does whatever Mac OS wants it to. iStat Menu doesn't even correctly display the fan percentage when I change the values with SilentStudio to under 1300rpm. For example, it's set to 900rpm when the temperature hits 60°C and iStat Menu just displays 0% despite correctly displaying the rpm. However, if I turn the off, it actually says so in the percentage item. 

I'll try it out on my MacBook Pro, too.

**Comment by rcambrj** [comment]
Just bought the pro version of this app and noticed that the fans don't go to zero when configured with a sensor-based range, only to find that there's a 4+ year old issue about it. That sucks.

**Comment by 13thvanger** [comment]
Maybe you could make a setting to switch to a non-auto preset and back depending on the temperature?
